telegram/ext/_commandhandler.py

```diff
@@ -155,6 +158,29 @@ def collect_additional_context(
             if isinstance(check_result[1], dict):
                 context.update(check_result[1])
 
+    def _parse_commands(self, command: SLTS[str]) -> None:
+        if isinstance(command, str):
+            self._commands = {command.lower()}
+        else:
+            self._commands = {i.lower() for i in command}
+
+    @property
+    def commands(self) -> Set[str]:
+        """
+        The set of commands this handler should listen for.
+
+        Returns:
+            Set[:obj:`str`]
+        """
+        return self._commands
+
+    @commands.setter
+    def commands(self, command: SLTS) -> None:
+        self._parse_commands(command)
+        for comm in self._commands:
+            if not re.match(r'^[\da-z_]{1,32}$', comm):
+                raise ValueError(f'Command `{comm}` is not a valid bot command')
```

Comment:
when this is raised, `self._commands` has already been changed - it should be the other way around. You can even add a unit test along the lines

1. bulid commandhandler with valid commands
2. try to assign invalid commands
3. assert that the commands where not changed

what we can do to make this easier here & also make the init a bit cleaner is:

1. make `_parse_commands` return the parsed commands instead of assigning them directly to `self._commands`. It can be converted into a static method
2. apply the regexing for valid commands directly within `_parse_commands`
3. do `self._commands = self._parse_commands(command)` both here and in `__init__`.